File: example-java/src/main/java/ru/service/MyService.java
package ru.service;

import ru.communicator.*;
import ru.model.*;
import static io.art.logging.Logging.*;
import static io.art.transport.extensions.TransportExtensions.*;

public class MyService implements MyCommunicator {
    @Override
    public Model myMethod(Model model) {
        logger(MyService.class).info("myMethod:\n" + toPrettyString(model));
        return model.toBuilder().value(model.getValue() + ": response").build();
    }

    @Override
    public Model getModel() {
        logger(MyService.class).info("getModel");
        return Model.builder().value("http: response").build();
    }
}

File: example-kotlin/src/main/kotlin/ru/service/MyService.kt
package ru.service

import io.art.core.kotlin.compensate
import io.art.logging.kotlin.info
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import ru.communicator.MyCommunicator
import ru.model.Model

object MyService : MyCommunicator {
    override fun myMethod(model: Model): Model {
        info("myMethod")

        return model.copy(value = sequenceOf("${model.value.first()}: response"))
    }

    override fun getModel(): Model = Model(sequenceOf("http: response")).apply { info("getModel") }

    override fun compensation(input: Flux<String>): Mono<String> = input
            .compensate({ value -> value == "error" }) { Flux.just("not error") }
            .last()
}
